aten/src/ATen/Device.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <ATen/Error.h>
+#include <ATen/ScalarType.h>
+#include <ATen/Tensor.h>
+#include <ATen/TensorMethods.h>
+#include <ATen/Type.h>
+#include <ATen/optional.h>
+
+#include <cstddef>
+#include <iosfwd>
+#include <string>
+
+namespace at {
+/// Represents a a compute device on which a tensor is located. A device is
+/// uniquely identified by a type, which specifies the type of machine it is
+/// (e.g. CPU or CUDA GPU), and a device index or ordinal, which identifies the
+/// specific compute device when there is more than one of a certain type. The
+/// device index is optional, and in its defaulted state represents (abstractly)
+/// "the current device". Further, there are two constraints on the value of the
+/// device index, if one is explicitly stored:
+/// 1. The device index is never negative.
+/// 2. When the device type is CPU, the device index must be zero.
```

Comment:
should this be..."when the device type is CPU and a device ordinal exists,it must be 0?